### DISCUSSION TITLE
Disable the bullmq module to consume jobs

### DISCUSSION THREAD

**Discussion by adrijere** [taskforcesh/bullmq#1829] [Q&A]
Hi 👋 

Thanks for your work!
I currently have an NestJS Application which uses my BullMQ Module. I'm creating a `worker.ts` to only consume jobs.
I've in mind to create a `dynamic module` and only allow my NestJS application to produce jobs and my worker to consume jobs.

I looked for an option to change in the `config` of BullMQ but found nothing so far.
Is there a way to handle this case with BullMQ or is there something in the roadmap to handle this?

Thanks a lot for your time and have a nice weekend!

**Comment by manast** [comment]
Not sure I understand what you mean. In BullMQ you can define workers as separate entities as many as you want.

**Comment by adrijere** [comment]
I mean, I want to let my main application to produce jobs and I want to have a dedicated instance to consume jobs as background jobs. For that, I'm creating a "worker" (my dedicated instance) as standalone application with `createApplicationContext`. Hope it's more clear now, sorry for the misunderstanding with the word `worker` which is also a keyword in `BullMQ` 🙈

**Reply by manast** [reply]
I am not familiar with NestJS, maybe somebody who knows more can fill in some details, but basically, it seems to be there should not be any problem in separating the producer of jobs from the consumer as it is how BullMQ is designed to be used.

**Comment by swellee** [comment]
hey, I guess you need this:
[nestjs with bullmq](https://docs.bullmq.io/bullmq-pro/nestjs)
